meson.build

```diff
@@ -24,6 +24,18 @@ endif
 dep_alsa = dependency('alsa', version: '>=1.0.18', required: alsa_required)
 dep_jack = dependency('jack')
 jack_version = dep_jack.version()
+jack_implementation = ''
+jack_implemetnation = dep_jack.get_variable('jack_implementation')
+if jack_implementation == 'pipewire'
+  message('Pipewire JACK implementation detected.')
+elif jack_implementation == 'jack2'
+  message('Jack2 JACK implementation detected.')
+elif jack_implementation == 'jack1'
+  message('Jack1 JACK implementation detected.')
+else
+  error('No compatible jack implementation found! Requires either jack1 >= 0.126.0, jack2 >= 1.9.20 or pipewire-jack >= 0.3.44.')
```

Comment:
this should be a warning, not an error.
if someone decides to do yet another jack server implementation, they should be able to build and use these tools as-is.

perhaps we should starting assuming jack2-like API usage if targeting anything other than jack1.

Reply:
Without an error we would need to start looking at very specific version ranges again (which is why I introduced the `jack_implementation` variable).
I don't want to assume anything but build based on the available features. The error is there, so that we can guarantee that a particular version is working as intended (i.e. jack-example-tools can be compiled with and installed side-by-side any of the mentioned jack implementations).

If there will be other jack implementations in the future, we can add them here and they should also expose `jack_implementation` via pkgconfig. Ideally this would only happen after adding them in CI though of course! :)